openupgrade_scripts/scripts/account/15.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,429 @@
+from openupgradelib import openupgrade
+
+
+def _convert_field_to_html(env):
+    openupgrade.convert_field_to_html(
+        env.cr, "res_company", "invoice_terms", "invoice_terms"
+    )
+    openupgrade.convert_field_to_html(env.cr, "account_fiscal_position", "note", "note")
+    openupgrade.convert_field_to_html(env.cr, "account_move", "narration", "narration")
+    openupgrade.convert_field_to_html(env.cr, "account_payment_term", "note", "note")
+
+
+def _fast_fill_account_move_always_tax_exigible(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_move
+        ADD COLUMN IF NOT EXISTS always_tax_exigible BOOLEAN""",
+    )
+    # 1. Set always_tax_exigible = False if record.is_invoice(True) is True
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move
+        SET always_tax_exigible = false
+        WHERE move_type IN (
+            'out_invoice',
+            'out_refund',
+            'in_refund',
+            'in_invoice',
+            'out_receipt',
+            'in_receipt')
+        """,
+    )
+    # 2. Set always_tax_exigible = False
+    #    if record._collect_tax_cash_basis_values() is True
+    # 2.1 Set always_tax_exigible = True if invoice is multiple involved currencies
+    #     else False
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move am
+        SET always_tax_exigible =
+            CASE
+                WHEN (SELECT COUNT(aml.currency_id)
+                     FROM account_move_line aml
+                     WHERE aml.move_id = am.id) > 1
+                THEN true
+                ELSE false
+            END
+        WHERE am.always_tax_exigible IS NULL""",
+    )
```

Comment:
```suggestion
    openupgrade.logged_query(
        env.cr,
        """
        UPDATE account_move am
        SET always_tax_exigible =
            CASE
                WHEN (SELECT COUNT(aml.currency_id)
                     FROM account_move_line aml
                     WHERE aml.move_id = am.id) > 1
                THEN true
                ELSE NULL
            END
        WHERE am.always_tax_exigible IS NULL""",
    )
```
Otherwise, the below queries will be useless